### PR TITLE
Automated Transfers

### LINKED ISSUE #7653: Automated Transfer v1 Checklist
- [x] Update the fluxc hash and handle the error codes for eligibility instead of just showing the error message
- [x] After AT is completed before fetching the site and its plugins, change the progress dialog to indeterminate and tell the user that we are wrapping things up [#](https://github.com/wordpress-mobile/WordPress-Android/pull/7609#discussion_r180238917)
- [x] If fetching the site or its plugins result in an error after AT is completed, handle it gracefully [#](https://github.com/wordpress-mobile/WordPress-Android/pull/7609#discussion_r180239942)
- [x] Handle the eligibility check `event.error` [#](https://github.com/wordpress-mobile/WordPress-Android/pull/7609#discussion_r180538904)
- [x] Add logs for AT
- [x] Design review
- [x] Copy check
- [x] Analytics
- [x] Feature flag (?): Probably not needed anymore, but adding it here as a reminder.
- [x] Merge `origin/develop` and update FluxC hash to latest: I am not doing this just yet because updating the FluxC hash to latest would break the build due to #7658 not being merged yet.

/cc @nbradbury

### PR DESCRIPTION
**Pull request by oguzkocer**
Fixes #7653. This PR adds the Automated Transfers feature. All the changes has been reviewed in #7609, #7661 & #7664 (including design, analytics and copy reviews). I've only merged `develop` into the branch after the last PR and ran the test below to make sure everything is still working as expected. It'd be great if the reviewer can do a test run as well before merging the code.

To test:
* Create a new WordPress.com site with business plan
* Add a custom domain to your site (A subdomain works as well, if you already have a domain, just map a subdomain of it)
* Make sure it’s public.
* Make sure you’re an admin.
* Go to My Site and check that Plugins is visible (if you have made a recent change in Calypso, you may need to go to site list and refresh the sites)
* Go into Plugins and tap on any plugin and try to install it
* Wait about a minute or so and then verify that “Successfully installed {plugin name}” snackbar shows up and the plugin shows up as installed.
* Go back to Plugins and make sure the site plugins section is now visible (Note that the backend will keep installing a few more plugins, so you may see more plugins show up in the next few minutes)

### CODE REVIEW

**Comment by oguzkocer** [comment]
It turns out I spoke too soon, there was a lint issue for specific imports which should be fixed in fde52fe5ffbec37720d6ae69dd662371216de5d1. If not, I'll update that commit to fix any other lint/checkstyle issues that come up. Sorry for that!

**Comment by 0nko** [comment]
Tested and verified, works as expected. Nice PR!

:shipit:

**Comment by designsimply** [comment]
<blockquote>Wait about a minute or so and then verify that “Successfully installed {plugin name}” snackbar shows up and the plugin shows up as installed.</blockquote>

Quick note to say that I tested this and was able to successfully install a plugin and that I did not see an explicit notice that said “Successfully installed {plugin name}” but I did see the "Installing {plugin name}" notice disappear and get replaced with version options for Active, Autoupdates, and Remove:

**Comment by Robertght** [comment]
First, nice job! :)

- Upgraded to Alpha
- Added a new site via App
- Had to go to the Web app to upgrade to Business and add a domain
- Returned to the App and tried to install a plugin. Notifications came up in the feed regarding the transfer process. 
- I switched by mistake to the notifications and when I returned to the Plugins, the install button for WooCommerce was active. It - kept saying that I cannot install plugins on this site.
- I gave it a minute to let everything sync and tried with another plugin, Yoast this time. It did install nicely without any errors.
- Returning to WooCommerce, it looks like it isn’t installed and I keep getting Error installing WooCommerce [Retry]. When going to wp-admin the plugin is installed and activate successfully.
- After returning to the wp-admin side the plugin showed as installed correctly.

To sum up, is there a way we can delay the user into doing anything to the website via the app while the other plugins are setting up in the background? I felt like everything went so fast and neat but I had a bit of a shutdown when I encountered the error messages which could cause confusion/frustration among users. 

I also noticed the user needs to get in the browser to complete the plan upgrade and to register a domain - my concern is with the flow and I know there might be some limitations which block us at the moment from doing everything in the same place.
